packages/link-ui/src/LinkMenu/LinkMenu.tsx

```diff
@@ -0,0 +1,166 @@
+import React, {useEffect, useState} from 'react'
+import {BaseRange, Editor, Element, BaseEditor} from 'slate'
+import {
+  getSlatePluginType,
+  useEventEditorId,
+  useStoreEditorState
+} from '@udecode/slate-plugins-core'
+import {
+  upsertLinkAtSelection,
+  validateUrl,
+  removeLink,
+  ELEMENT_LINK
+} from '@dreifuss-wysiwyg-editor/slate-plugins-link'
+import './link.css'
+
+import {ReactEditor} from 'slate-react'
+import {HistoryEditor} from 'slate-history'
+
+type CustomElement = {type: 'link'; title: string; url?: string; children: CustomText[]}
+type CustomText = {title: string; url?: string; text: string}
+
+declare module 'slate' {
+  interface CustomTypes {
+    Editor: BaseEditor & ReactEditor & HistoryEditor
+    Element: CustomElement
+    Text: CustomText
+  }
+}
+
+export const ToolbarLink = () => {
+  const editor = useStoreEditorState(useEventEditorId('focus'))
+
+  const [title, setTitle] = useState('')
+  const [url, setURL] = useState('')
+
+  enum prefixType {
+    http = 'http://',
+    https = 'https://',
+    mailto = 'mailto://',
+    other = 'other'
+  }
+
+  const [prefix, setPrefix] = useState<prefixType | string>(prefixType.http)
+
+  const [selection, setSelection] = useState<BaseRange | null>(null)
+
+  const [isValidURL, setIsValidURL] = useState(false)
+  const isDisabled = !url || !title
```

Comment:
i think this should be a react state as well and you might need another useEffect that listens to `url`, `title` and `isValidUrl`.